packages/rfw/test/runtime_test.dart

```diff
@@ -1173,6 +1178,84 @@ void main() {
       expect(tester.widget<Text>(textFinder).data, '1 + 2 = 3');
     });
 
+    testWidgets('Widget builders - work with state', (WidgetTester tester) async {
+      final Widget widget = _setUp('''
+        import core;
+        import local;
+
+        widget test {counter: 0} = Calculator(
+          operand1: state.counter,
+          operand2: 1,
+          operation: 'sum',
+          builder: (result) => CoolText(
+            text: ['Counter: ', result.result],
+            onPressed: set state.counter = result.result,
+          ),
+        );
+      ''');
+      await tester.pumpWidget(widget);
+
+      final Finder textFinder = find.byType(Text);
+      expect(textFinder, findsOneWidget);
+      expect(tester.widget<Text>(textFinder).data, 'Counter: 1');
+
+      await tester.tap(textFinder);
+      await tester.pump();
+      expect(tester.widget<Text>(textFinder).data, 'Counter: 2');
+
+      await tester.tap(textFinder);
+      await tester.pump();
+      expect(tester.widget<Text>(textFinder).data, 'Counter: 3');
+    });
+
+    testWidgets('Widget builders - work with data', (WidgetTester tester) async {
+      final Widget widget = _setUp('''
+        import core;
+        import local;
+
+        widget test {counter: 0} = Calculator(
+          operand1: state.counter,
+          operand2: data.increment,
+          operation: 'sum',
+          builder: (result) => CoolText(
+            text: ['Counter: ', result.result],
+            onPressed: set state.counter = result.result,
+          ),
+        );
+      ''');
+      await tester.pumpWidget(widget);
+
+      final Finder textFinder = find.byType(Text);
+
+      data.update('increment', 1);
+      await tester.pump();
+      expect(textFinder, findsOneWidget);
+      expect(tester.widget<Text>(textFinder).data, 'Counter: 1');
+
+      await tester.tap(textFinder);
```

Comment:
can we check that it also updated the rendering without the tap?

Reply:
Sure thing, done!